test/Microsoft.AspNetCore.Mvc.Formatters.Json.Test/Internal/JsonResultExecutorTest.cs

```diff
@@ -156,6 +157,31 @@ public class JsonResultExecutorTest
             Assert.Equal("application/json; charset=utf-8", context.HttpContext.Response.ContentType);
         }
 
+        [Fact]
+        public async Task ExecuteAsync_ErrorDuringSerialization_DoesNotCloseTheBrackets()
+        {
+            // Arrange
+            var expected = Encoding.UTF8.GetBytes("{\"name\":\"Robert\"");
+
+            var context = GetActionContext();
+
+            var result = new JsonResult(new ModelWithSerializationError());
+            var executor = CreateExcutor();
+
+            // Act
+            try
+            {
+                await executor.ExecuteAsync(context, result);
+            }
+            catch
```

Comment:
Know I already approved but&hellip; This is too broad and doesn't fail when _no_ exception is thrown. Change this to `Assert.ThrowsAsync<TSpecificException>(() => executor.ExecuteAsync(...));`. Same in the other new test.